### PR TITLE
[ADAM-600] Adding RegionJoin trait.

### LINKED ISSUE #600: Consolidate join functionality via a trait
This was discussed in #517. Instead of layering the functionality via an additional `RegionRDDFunctions` layer, @laserson proposed layering this on top of the various region join implementations via a trait.

It might also make sense to implement some sort of cardinality estimation to select a join strategy, but that should probably be follow on work.

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #600. @tdanford @laserson thoughts about this?

### CODE REVIEW

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/RegionJoin.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.apache.spark.SparkContext._
+import scala.Predef._
+import org.apache.spark.SparkContext
+import scala.reflect.ClassTag
+
+trait RegionJoin {
+  /**
+   * Performs a region join between two RDDs.
+   *
+   * @param sc A SparkContext for the cluster that will perform the join
```
Comment (laserson) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/RegionJoin.scala]: delete this?

**Comment by laserson** [comment]
+1.  I'll wait for possible comments before merging

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/RegionJoin.scala (hunk 1 above)
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/RegionJoin.scala]: Thanks; fixing...

**Comment by laserson** [comment]
Thanks @fnothaft!

**Pull request merged**